File: schedule.java
//University of Central Florida
//16th Annual High School Programming Tournament
//May 3rd, 2002
//
//Problem Name: Conflicting Conferences
//Filename: schedule.java
//Input File: schedule.in

import java.io.*;
import java.lang.*;
import java.util.*;

class schedule {

public static void main (String [] args)
{
	int num_of_datasets;
	String line;

	//we use this array to keep track of any conflicts. If the number
	//at any given index i is greater than 1, then that means at hour i,
	//the room has a conflict.
	int[] roomused = new int[24];

	try
	{
		//open the input files.
		FileReader fr = new FileReader("schedule.in");
		BufferedReader in = new BufferedReader(fr);

		//read in the number of datasets
		num_of_datasets = Integer.parseInt( in.readLine() );

		//loop over those datasets
		while( num_of_datasets-- >0 )
		{
			int r,conflicts,i;
			int start,end;

			conflicts=0;

			//print out the name of the room
			System.out.println( in.readLine() );

			//initialize the roomsused array so that no rentals
			//have yet occured
			for(i=0;i<24;i++)
				roomused[i]=0;

			//find out how many reservations there are to read in
			r = Integer.parseInt( in.readLine() );

			//loop over that number of reservations
			while(r-->0)
			{
				//we need to tokenize out the starting and
				//ending times
				StringTokenizer tk = new StringTokenizer( in.readLine() );
				start = Integer.parseInt( tk.nextToken() );

				end = Integer.parseInt( tk.nextToken() );

				//mark the room as rented for these times...
				for(i=start;i<end;i++)
				{
					roomused[i]++;
					if(roomused[i]>1)
						//conflicts is a flag to let
						//us know if any conflicts exist
						conflicts=1;
				}
			}

			//if there are conflicts...
			if( conflicts>0 )
			{
				System.out.print("CONFLICTS AT ");
				for(i=0;i<24;i++)
				{
					if( roomused[i]>1)
					{
						//if the room is rented out more
						//than once at a given hour...

						System.out.print( "" + i + "-" + (i+1) + " " );
					}
				}
				System.out.println();
				System.out.println();
			}
			else
			{
				//if there are no conflicts...
				System.out.println("SCHEDULE OK!");
				System.out.println();
			}
		}

	}
	//misc. exception stuff...
	catch ( FileNotFoundException e )
	{
		System.out.println( "couldn't open input" );
	}	
	catch( IOException e )
	{
		System.out.println( "some IO Exception");
	}

}

}

File: schedule.c
//University of Central Florida
//16th Annual High School Programming Tournament
//May 3rd, 2002
//
//Problem Name: Conflicting Conferences
//Filename: schedule.c
//Input File: schedule.in

#include <stdio.h>

int main()
{
	//we store how many rentals are occuring at any given hour
	//if the number is greater than 1, that means a conflict has
	//occured
	int roomused[24];

	//this is used as a buffer for the input data
	char line[300];

	int i;
	int num_of_datasets;

	//open the input file and redirect is as keyboard input
	freopen("schedule.in","r",stdin);

	//read in the number of datasets (WARNING: gets is VERY DANGEROUS in
	// production software! but there aren't any hackers here...)
	gets(line);
	sscanf(line,"%d",&num_of_datasets);

	//loop over the number of datasets
	while( num_of_datasets-- )
	{
		int r;
		int conflicts=0;

		//read in and then print out the room name
		gets(line);	//gets strips the "\n" for us
		printf("%s\n",line);

		//initialize the detect conflicts table
		for(i=0;i<24;i++)
			roomused[i]=0;

		//find out how many rentals there are
		gets(line);
		sscanf(line,"%d",&r);

		//loop over that number of rentals (will loop until r==0)
		while(r--)
		{
			int start,end;

			//read in the start and end times
			gets(line);
			sscanf(line,"%d %d", &start, &end );

			//mark the initialize table that the room is
			//rented at these times
			for(i=start;i<end;i++)
			{
				roomused[i]++;
				if(roomused[i]>1) 
					//set a flag that conflicts exist
					conflicts=1;
			}
		}

		//if there are conflicts, print them out
		if( conflicts )
		{
			printf("CONFLICTS AT ");
			for(i=0;i<24;i++)
			{
				if( roomused[i]>1 )
					//if the room is rented out more than once for this time period...
					printf("%d-%d ", i, i+1 );	//end letter
			}
			printf("\n\n");
		}	
		else	//if there are no conflicts...
			printf("SCHEDULE OK!\n\n");
	}
}
